### PR TITLE
Rename "portalIP" to "clusterIP"

### PR DESCRIPTION
**Pull request by thockin**
The term "portal" is confusing to people - we don't need another term for a concept that is closely aligned to one we already know - VIP.  

API: Rename the 'portalIP' field to 'clusterIP' in v1 API (v1beta3 and earlier are unchanged)
Flags: Alias the --portal-net flag (on apiserver) to --service-cluster-ip-range.  Old flag will continue to work.
Turnup: Rename the PORTAL_NET variable to SERVICE_CLUSTER_IP_RANGE

### CODE REVIEW

**Comment by bgrant0607** [comment]
Looking good already. Thanks much.

**Comment by justinsb** [comment]
I like this; it will be much more consistent.

**Comment by thockin** [comment]
This depends on the services doc update.

Removed WIP - I think it is done.  e2e passes (except PD, grr)

**Comment by thockin** [comment]
FWIW I find it word y in places but it is consistent.

**Comment by thockin** [comment]
@zmerlynn @roberthbailey @cjcullen regarding master-run addons and config params in general

**Comment by roberthbailey** [comment]
Does this require a new kube2sky image to be built once the change goes in on the master side?

**Comment by bgrant0607** [comment]
Please write a PR description that can be used in the release notes. The flag changes, in particular, are important to anyone whose turnup code isn't in our repo.

File: docs/services.md [hunk 1]
```diff
@@ -128,15 +138,15 @@ Accessing a `Service` without a selector works the same as if it had selector.
 The traffic will be routed to endpoints defined by the user (`1.2.3.4:80` in
 this example).
 
-## Portals and service proxies
+## Virtual IPs and service proxies
```
Comment (bgrant0607) [inline docs/services.md]: Possible ways to relate VIPs and cluster IPs:
* Virtual cluster IPs
* Virtual IPs (aka cluster IPs)
* Cluster-local virtual IPs

**Comment by bgrant0607** [comment]
Does this subsume #8655?

**Comment by bgrant0607** [comment]
@roberthbailey This only changes the v1 API, so kube2sky, other add-ons, configs, examples, etc. don't need to be updated until we move them to v1.

File: docs/services.md [hunk 2]
```diff
@@ -208,70 +268,109 @@ DNS records for each.  If DNS has been enabled throughout the cluster then all
...
-services, a virtual IP is not allocated, DNS is not configured (this will be
-fixed), and service-specific environment variables for pods are not created.
-Additionally, the kube proxy does not handle these services and there is no
-load balancing or proxying done by the platform for them. The endpoints
-controller will still create endpoint records in the API for such services.
-These services also take advantage of any UI, readiness probes, etc. that are
-applicable for services in general.
+Sometimes you don't need or want load-balancing and a single virtual IP.  In
+this case, you can create "headless" services by specifying `"None"` for the
+cluster IP (`spec.clusterIP` or `spec.portalIP` in v1beta3 and earlier APIs).
```
Comment (bgrant0607) [inline docs/services.md]: Needs a comma before "or spec.portalIP".

File: docs/services.md [hunk 3]
```diff
@@ -208,70 +268,109 @@ DNS records for each.  If DNS has been enabled throughout the cluster then all
...
-address and difficult to re-configure.  The `PortalIP` address that a user
-chooses must be a valid IP address and within the portal net CIDR range that is
-specified by flag to the API server.  If the PortalIP value is invalid, the
-apiserver returns a 422 HTTP status code to indicate that the value is invalid.
+address.  Kubernetes supports two ways of doing this: `NodePort`s and
+`LoadBalancer`s.
+
+Every `Service` has a `Type` field which defines how the `Service` can be
+accessed.  Valid values for this field are:
+   - ClusterIP: use a cluster-internal VIP (portal) only - this is the default
```
Comment (bgrant0607) [inline docs/services.md]: I think portal could be dropped here.

**Comment by bgrant0607** [comment]
Looks pretty good overall.

**Comment by thockin** [comment]
PR description (first comment) updated.

Style question - is "ClusterIP" a noun or is it "Cluster IP".  Ditto for "NodePort" vs "Node Port" ?

**Comment by bgrant0607** [comment]
We're in no way consistent about this, but I'd reserve compound CamelCase (e.g., `ReplicationController`) for cases where you are referring to the code (fields, kinds, etc.) and just use separate words (e.g., Replication Controller) in most descriptions.

**Comment by bgrant0607** [comment]
LGTM. Please merge #8655 first, which I expect to cause this to need to be rebased.

**Comment by thockin** [comment]
Rebased and pushed

**Pull request merged**